lcls-twincat-motion/Library/DUTs/DUT_PositionState.TcDUT

```diff
@@ -1,46 +1,71 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4022.12">
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4022.18">
   <DUT Name="DUT_PositionState" Id="{85d45494-8a84-426c-8efe-9136ee5821c6}">
     <Declaration><![CDATA[TYPE DUT_PositionState :
-// Include io: input on things that should never be mutable in EPICS.
-// Omit io on things that may or may not need to be mutable, allow engineer to decide in their pragma.
-// Currently: some installations would want position and delta to be mutable.
+	// Defines settings and current safety status for moves to specific positions for an axis
 STRUCT
+	// Name as queried via the NAME PV in EPICS
 	{attribute 'pytmc' := '
 		pv: NAME
 		io: input
 	'}
-	sName: STRING;
+	sName: STRING := 'Invalid';
 
+	// Position associated with this state
 	{attribute 'pytmc' := '
 		pv: SETPOINT
 		io: io
 	'}
 	fPosition: LREAL;
 	
+	// Maximum allowable deviation from fPosition while at the state 
 	{attribute 'pytmc' := '
 		pv: DELTA
 		io: io
 	'}
 	fDelta: LREAL;
```

Comment:
fDelta needs to be positive. `field: DRVL 0.0` perhaps?

Reply:
You're right. I should also take the absolute value of this parameter in the implementation of `F_AtPositionState` to guard against PLC code setting it to a negative value.